server.go

```diff
@@ -2139,7 +2140,9 @@ func (s *server) Start() error {
 				peerAddr, true,
 				s.cfg.ConnectionTimeout,
 			)
-			if err != nil {
+			if err != nil &&
```

Comment:
Seems this would only happen if `AddPeers` had a duplicate? Or I guess if two DNS hosts return the same IP?

Reply:
I think it might be that the connection manager started just above accepted a connection from the configured peer before the attempt to add the peer from the config here. Unfortunately I can't get to the logs from before applying this fix any more to prove that this is what happened but I'll add an itest for this to make sure I'm going in the right direction. Thanks!